src/rust/src/ed25519.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+use ed25519_zebra::{Signature, SigningKey, VerificationKey};
+use libc::{c_uchar, size_t};
+use rand_core::OsRng;
+use std::convert::TryFrom;
+use std::slice;
+
+#[no_mangle]
+pub extern "C" fn rust_ed25519_generate_keypair(sk: *mut [u8; 32], vk: *mut [u8; 32]) {
+    let signing_key = SigningKey::new(OsRng);
+
+    if let Some(sk) = unsafe { sk.as_mut() } {
+        *sk = signing_key.into();
+    }
```

Comment:
This does nothing if the pointer is null. We should probably abort instead. Similarly below.